server/server.go

```diff
@@ -223,60 +223,59 @@ func (s *Server) githubEvent(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	var pingEvent *github.PingEvent
 	if r.Header.Get("X-GitHub-Event") == "ping" {
-		pingEvent = PingEventFromJSON(ioutil.NopCloser(bytes.NewBuffer(buf)))
+		if pingEvent := PingEventFromJSON(ioutil.NopCloser(bytes.NewBuffer(buf))); pingEvent != nil {
+			mlog.Info("ping event", mlog.Int64("HookID", pingEvent.GetHookID()))
+			return
+		}
 	}
 
 	event := PullRequestEventFromJSON(ioutil.NopCloser(bytes.NewBuffer(buf)))
-	eventIssueComment := IssueCommentFromJSON(ioutil.NopCloser(bytes.NewBuffer(buf)))
-
 	if event != nil && event.PRNumber != 0 {
 		mlog.Info("pr event", mlog.Int("pr", event.PRNumber), mlog.String("action", event.Action))
 		s.handlePullRequestEvent(ctx, event)
 		return
 	}
 
-	pr, err := s.getPRFromComment(ctx, *eventIssueComment)
+	eventData := EventDataFromJSON(ioutil.NopCloser(bytes.NewBuffer(buf)))
+	if eventData == nil || eventData.Action != "created" {
```

Comment:
This is confusing :) We are checking for something else, and passing something else in the method. Can't wait to refactor this.

To be extra sure, can we do some manual testing of the 3 cases (handlePullRequestEvent/handleIssueEvent/rest) by constructing some curl requests and checking that the correct code paths are being followed? This looks okay from code, but without any actual tests, I am not really confident.

Reply:
Agree with you on the confusion, this will hopefully be lost in dusty pages of history pretty soon. As for the `curl` requests, with simple data we can confirm high level paths but it will going to fail (panic) eventually since we need a real data to continue on communication with GitHub. Otherwise I may need some real GH requests to test these.

Reply:
> but it will going to fail (panic) eventually since we need a real data to continue on communication with GitHub. 

Why would it panic? I would assume that the request will fail because it's not authenticated.

Reply:
You're right, it would fail before it panics. So this needs to be tested in a staging env.?

Reply:
There's no staging env yet. I was thinking of just manually testing with curl. The request can fail eventually. But I wanted to verify that the right code paths are being taken.